server/platform/shared/filestore/s3store_test.go

```diff
@@ -320,3 +321,58 @@ func (fc fauxCloser) Close() error {
 	fc.s3WithCancel.cancel()
 	return fc.closeErr
 }
+
+func TestListDirectory(t *testing.T) {
+	s3Host := os.Getenv("CI_MINIO_HOST")
+	if s3Host == "" {
+		s3Host = "localhost"
+	}
+
+	s3Port := os.Getenv("CI_MINIO_PORT")
+	if s3Port == "" {
+		s3Port = "9000"
+	}
+
+	s3Endpoint := fmt.Sprintf("%s:%s", s3Host, s3Port)
+
+	cfg := FileBackendSettings{
+		DriverName:                         driverS3,
+		AmazonS3AccessKeyId:                "minioaccesskey",
+		AmazonS3SecretAccessKey:            "miniosecretkey",
+		AmazonS3Bucket:                     "mattermost-test-1",
+		AmazonS3Region:                     "",
+		AmazonS3Endpoint:                   s3Endpoint,
+		AmazonS3PathPrefix:                 "",
+		AmazonS3SSL:                        false,
+		AmazonS3SSE:                        false,
+		AmazonS3RequestTimeoutMilliseconds: 5000,
+	}
+
+	fileBackend, err := NewS3FileBackend(cfg)
+	require.NoError(t, err)
+
+	found, err := fileBackend.client.BucketExists(context.Background(), cfg.AmazonS3Bucket)
+	require.NoError(t, err)
+
+	if !found {
+		err = fileBackend.MakeBucket()
+		require.NoError(t, err)
+	}
+
+	fileBackend.pathPrefix = "19700101/"
+	require.NoError(t, err)
+	b := []byte("test")
+
+	path1 := "19700101/" + randomString() + ".txt"
+	_, err = fileBackend.WriteFile(bytes.NewReader(b), path1)
+	require.NoError(t, err, "Failed to write file1 to S3")
+
+	_, err = fileBackend.listDirectory("", false)
+	require.Error(t, err, "readdir 19700101/: file does not exist []")
```

Comment:
Check for the actual error type apart from just asserting for the error. 

```
var pErr *fs.PathError
assert.True(t, errors.As(err, &pErr), "error is not of type fs.PathError")
```

Reply:
sure